### PR TITLE
Watchdog no longer automatically rewrites every test, also not on Windows

### LINKED ISSUE #352: `test-watchdog_testfiles.R` Issue
Hey hey @timtreis !

I wanted to let you know that when I run the unit tests, the `test-watchdog_testfiles.R` file deletes all the unit testing, rewrites then with a new header, but also adds additional spaces between lines every time the tests are run. If I forget to skip these tests locally, each unit test file grows very quickly. Also, the re-writing makes git think I modified every line of every unit testing file. I am not sure the purpose of this file and how it works with the unique unit testing setup of the package.

### PR DESCRIPTION
**Pull request by timtreis**
**What changes are proposed in this pull request?**

This PR changes the watchdog to only run if the timestamp in the header is "outdated", meaning that the file has been changed since the watchdog last ran. Also, I'm skipping the test on Windows for now since I don't know why @ddsjoberg gets double-newlines everywhere.

--------------------------------------------------------------------------------

Checklist for PR reviewer

- [ ] PR branch has pulled the most recent updates from main branch. Ensure the pull request branch and your local version match and both have the latest updates from the main branch.
- [ ] If a new function was added, function should be included in `_pkgdown.yml`
- [ ] If a bug was fixed, a unit test was added for the bug check
- [ ] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [ ] Code coverage is suitable for any new functions/features. Review coverage with `withr::with_envvar(new = c("NOT_CRAN" = "true"), covr::report())`. Before you run, begin a fresh R session without any packages loaded. 
- [ ] R CMD Check runs without errors, warnings, and notes
- [ ] `usethis::use_spell_check()` runs with no spelling errors in documentation
- [ ] Has `NEWS.md` been updated with the changes from this pull request under the heading indicating the latest version. If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] Has the version number been incremented using `usethis::use_version(which = "dev")` 
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

File: tests/testthat/test-get_risktable.R [hunk 1]
```diff
@@ -27,7 +27,7 @@
 #' T4.1 The function supplies defaults to increase the length of the `label` vector to same length as the `statlist` vector
 #' T4.2 The supplied defaults for the `label` vector match the arguments specified in the `statlist`
 #' T4.3 The function limits the length of the `label` vector to the length of the `statlist` vector
-#' T5. The functions calculates requested summary across time
+#' T5. T5. The functions calculates requested summary across time
```
Comment (SHAESEN2) [inline tests/testthat/test-get_risktable.R:30]: duplicate T5.

File: tests/testthat/test-get_risktable.R [hunk 2]
```diff
@@ -47,13 +47,13 @@
...
 #' T7.4 The calculations are in agreement with expectations when grouped overall
 #' T7.5 No error when there is only one strata available and collapse = TRUE
-#' T8. The output dataset is a data.frame with attributes for downstream processing
-#' T8.1 The output dataset is a data.frame
-#' T8.2 The output dataset has the attribute `time_ticks` that specifies the times
-#' T8.3 The output dataset has the attribute `title` that specifies the labels used in downstream functions
-#' T8.4 The output dataset has the attribute `statlist` that reflects the ´group´ used
-#' T9. Tests for `get_risktable.tidycmprsk()`
-#' T9.1 Results are accurate without error
+#' T7. The output dataset is a data.frame with attributes for downstream processing
```
Comment (SHAESEN2) [inline tests/testthat/test-get_risktable.R:50]: Should be T8. I recently fixed this in a PR but cant remember which one.

File: tests/testthat/test-tidyme.R [hunk 3]
```diff
@@ -14,7 +14,7 @@
 #' T3.2 The S3 method, associated with a `survfit` object, has columns representing all list elements of the S3 object
 #' T3.3 The S3 method, associated with a `survfit` object, turns list elements that represent integer numbers into integers
 #' T3.4 The S3 method, assocated with a `survfit` object, turns the strata into a factor
-#' T3.5 The S3 method, associated with a `survfit` object, add the original object as an attribute to the tidied object
+#' testthat::test_that( T3.5 The S3 method, associated with a `survfit` object, add the original object as an attribute to the tidied object
```
Comment (SHAESEN2) [inline tests/testthat/test-tidyme.R:17]: fix

**Review by SHAESEN2 — CHANGES_REQUESTED** [review]

File: tests/testthat/test-watchdog_testfiles.R [hunk 4]
```diff
@@ -12,6 +12,7 @@ context("validate_watchdog - T1. Details on last change for test files are recor
 # skip on GH Actions and CRAN
 testthat::skip_if(isTRUE(as.logical(Sys.getenv("CI"))))
 testthat::skip_on_cran()
+testthat::skip_on_os("windows")
```
Comment (ddsjoberg) [inline tests/testthat/test-watchdog_testfiles.R:15]: thank you for adding! :)

**Comment by ddsjoberg** [comment]
Thank you for this update @timtreis !

Going forward if someone on a Windows machine submits a PR, would you like to see them modify the testing file header manually of any file they update?

I don't know how much I can test here, since it's skipped on my Windows machine 😝

**Pull request closed without merging**